src/auto-trace.helper.spec.js

```diff
@@ -60,6 +60,31 @@ describe('auto-trace.js', () => {
 			expect(result.autoTraceIgnore).toEqual(true);
 			expect(err.stack.indexOf('wrapObjectWithError')).toEqual(-1);
 		});
+		it('should store sync and async stack traces', () => {
+			const err = new Error('My sync error message will be preserved');
+			err.stack = `Error: My sync error message will be preserved
+	at catchAsyncStacktrace(/global-settings.js:1929:28)
+	at _showMoreLicensesDialog(/4.global-settings.js:428:111)
+	at HTMLUnknownElement.d(/static/raven/raven-3.9.1-d.min.js:2:6222)
+	at Array.forEach(<anonymous>)`;
+			const asyncErr = new Error('My async error message will live on in the stack');
+			asyncErr.stack = `Error: My async error message will live on in the stack
+	at catchAsyncStacktrace(./node_modules/auto-trace/lib/auto-trace.js:67:0)
+	at SigningModal._this.createOrGetSigningExperience(./src/signing-modal.component.js:110:4)
+	at createOrGetSigningExperience(./src/signing-modal.component.js:77:7)
+	at closeAll(../jspm_packages/npm/react-dom@15.5.4/lib/Transaction.js:153:15)`
+			const result = wrapObjectWithError(err, asyncErr);
+			const expectedStackTrace = `Error: My async error message will live on in the stack
+	at SigningModal._this.createOrGetSigningExperience(./src/signing-modal.component.js:110:4)
+	at createOrGetSigningExperience(./src/signing-modal.component.js:77:7)
+	at closeAll(../jspm_packages/npm/react-dom@15.5.4/lib/Transaction.js:153:15)
+  at AUTO TRACE SYNC: Error: My sync error message will be preserved
+	at _showMoreLicensesDialog(/4.global-settings.js:428:111)
+	at HTMLUnknownElement.d(/static/raven/raven-3.9.1-d.min.js:2:6222)
+	at Array.forEach(<anonymous>)`;
+			expect(result.stack).toEqual(expectedStackTrace);
+			expect(result.message).toEqual(`My sync error message will be preserved`);
+		});
```

Comment:
Creates a `async` and `sync` error. 

Checks that the stacktrace is `async` followed by `sync` stacktrace

Checks that the sync error message is preserved